src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  let anotherState = { ...state };
+  const massState = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    if (type === 'addProperties') {
+      Object.assign(anotherState, extraData);
+    }
+
+    if (type === 'removeProperties') {
+      for (const del of keysToRemove) {
+        delete anotherState[del];
+      }
+    }
 
+    if (type === 'clear') {
+      anotherState = {};
+    }
```

Comment:
The 'switch' statement should include a 'default' case to handle unexpected action types and provide error handling.